pkg/kotsadm/minio.go

```diff
@@ -115,3 +119,162 @@ func ensureMinioService(namespace string, clientset *kubernetes.Clientset) error
+		log.Info("")
+		log.Info("---TO INSTALL the local-volume-provider plugin---")
+		log.Info("For existing cluster installations, complete the installation instructions here:")
+		log.Info("https://github.com/replicatedhq/local-volume-provider")
+		log.Info("")
+		log.Info("If you're seeing this message on an embedded cluster installation, please contact your vendor for support.")
+		log.Info("")
+		log.Info("After the plugin has been installed, re-run your upgrade command.")
+
+		return errors.New("velero local-volume-provider plugin is not installed - cannot perform migration")
+	}
+
+	bsl, err := snapshot.FindBackupStoreLocation(context.TODO(), deployOptions.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to find backupstoragelocations")
+	}
+	if bsl == nil {
+		log.Info("Did not detect active Minio snapshot install. Cleaning up.")
+		// Minio FS is not longer enabled; clean up and exit
+		// Not catching the error here, as this is not required to proceed
+		snapshot.DeleteFileSystemMinio(context.TODO(), deployOptions.Namespace)
+		return nil
+	}
+
+	store, err := snapshot.GetGlobalStore(context.TODO(), deployOptions.Namespace, bsl)
+	if err != nil {
+		return errors.Wrap(err, "failed to get snapshot store")
+	}
+	if store == nil || store.FileSystem == nil {
+		log.Info("Did not detect active Minio snapshot install. Cleaning up.")
+		// Minio FS is not longer enabled; clean up and exit
+		// Not catching the error here, as this is not required to proceed
+		snapshot.DeleteFileSystemMinio(context.TODO(), deployOptions.Namespace)
+		return nil
+	}
+
+	previousBsl := bsl.DeepCopy()
+
+	previousBackups, err := snapshot.ListAllBackups(context.TODO(), snapshot.ListInstanceBackupsOptions{Namespace: deployOptions.Namespace})
+	if err != nil {
+		return errors.Wrap(err, "failed to list existing backups")
+	}
+
+	// Add the config map to configure the new plugin
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return errors.Wrap(err, "failed to get k8s clientset")
+	}
+	fsDeployOptions := &snapshot.FileSystemDeployOptions{
+		Namespace:        deployOptions.Namespace,
+		IsOpenShift:      k8sutil.IsOpenShift(clientset),
+		ForceReset:       false,
+		FileSystemConfig: *prevFsConfig,
+	}
+	err = snapshot.EnsureLocalVolumeProviderConfigMap(*fsDeployOptions, veleroNamespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to ensure velero local-volume-provider config map")
+	}
+
+	registryOptions, err := GetKotsadmOptionsFromCluster(deployOptions.Namespace, clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to get registry options from cluster")
+	}
+
+	start := metav1.Now()
+	success := false
+	defer func() {
+		if !success {
+			err := snapshot.RevertToMinioFS(context.TODO(), deployOptions.Namespace, veleroNamespace, previousBsl)
+			if err != nil {
+				log.Error(errors.Wrap(err, "Could not restore minio backup storage location"))
+			}
+			log.Info("Minio backup storage location restored")
```

Comment:
feels like this needs to be in an `else` statement

Reply:
yeah, or a `return`